lib/membrane/rtp/session_bin.ex

```diff
@@ -661,47 +682,52 @@ defmodule Membrane.RTP.SessionBin do
         {nil, state}
       end
 
-    maybe_link_twcc =
-      if should_link? do
-        fn link_builder ->
-          link_builder
-          |> via_in(Pad.ref(:input, pad_ssrc))
-          |> then(fn link ->
-            if should_create_child? do
-              to(link, :twcc_receiver, %{maybe_twcc | feedback_sender_ssrc: maybe_twcc_ssrc})
-            else
-              to(link, :twcc_receiver)
-            end
-          end)
-          |> via_out(Pad.ref(:output, pad_ssrc))
-        end
+    to_twcc_receiver = fn link ->
+      if should_create_child? do
+        to(link, :twcc_receiver, %{maybe_twcc | feedback_sender_ssrc: maybe_twcc_ssrc})
       else
-        & &1
+        to(link, :twcc_receiver)
       end
+    end
+
+    link_twcc = fn link_builder ->
+      link_builder
+      |> via_in(Pad.ref(:input, pad_ssrc))
+      |> then(to_twcc_receiver)
+      |> via_out(Pad.ref(:output, pad_ssrc))
+    end
+
+    maybe_link_twcc = if should_link?, do: link_twcc, else: & &1
 
     {rtp_extensions, maybe_link_twcc, state}
   end
 
-  defp maybe_handle_twcc_sender(pad_ssrc, ctx) do
+  defp maybe_handle_twcc_sender(rtp_extensions, pad_ssrc, ctx) do
     # Workaround: as TWCC is a transport-wide extension, there should exist only one TWCC sender
-    # child that handles packets for all outgoing streams. As there is no support for declaring
-    # outbound extensions, outbound TWCC will be enabled if inbound TWCC has been enabled.
-    should_link? = Map.has_key?(ctx.children, :twcc_receiver)
+    # child that handles packets for all outgoing streams.
+    maybe_twcc = Keyword.get(rtp_extensions, :twcc)
+
+    should_link? = maybe_twcc != nil
+
     should_create_child? = not Map.has_key?(ctx.children, :twcc_sender)
 
-    if should_link? do
-      fn link_builder ->
-        link_builder
-        |> via_in(Pad.ref(:input, pad_ssrc))
-        |> then(fn link ->
-          if should_create_child? do
-            to(link, :twcc_sender, RTP.TWCCSender)
-          else
-            to(link, :twcc_sender)
-          end
-        end)
-        |> via_out(Pad.ref(:output, pad_ssrc))
+    to_twcc_sender = fn link ->
+      if should_create_child? do
+        to(link, :twcc_sender, maybe_twcc)
+      else
+        to(link, :twcc_sender)
       end
+    end
```

Comment:
I had to move this out of the `link_twcc` as Credo was complaining about too deep nesting :thinking: